### PR TITLE
<DropDown /> Phase 1 - Clickable Label with basic design

### PR DESCRIPTION
**Pull request by liorzisman**
resolves issue #158

### CODE REVIEW

File: demo/components-demo.tsx [hunk 1]
```diff
@@ -19,6 +20,10 @@ export class ComponentsDemo extends React.Component<{}, {}> {
         return (
             <div>
                 <div>
+                    <h2>DropDown</h2>
+                    <DropDownDemo />
+                </div>
```
Comment (alisey) [inline demo/components-demo.tsx]: There should be a `<hr />` after the div.

File: src/components/drop-down/drop-down.st.css [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    font-size: 16px;
+}
+
+.drop-down-label {
+    border: 1px solid #adbac5;
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+
+    .caret {
```
Comment (alisey) [inline src/components/drop-down/drop-down.st.css]: Is there a reason for these rules to be nested in `.drop-down-label`?

Comment (liorzisman) [inline src/components/drop-down/drop-down.st.css]: might be a confusing name but its actually for the whole `<DropDownLabel />` component that contains the label string and and the `caret`

File: src/components/drop-down/drop-down.st.css [hunk 3]
```diff
@@ -0,0 +1,27 @@
+.root {
+    width: 276px;
```
Comment (liorzisman) [inline src/components/drop-down/drop-down.st.css]: change this

File: src/components/drop-down/drop-down.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import { SBComponent, SBStateless } from 'stylable-react-component/dist/stylable-react';
+import { CaretDown } from './drop-down-icons';
+import style from './drop-down.st.css';
+
+export interface DropDownLabelProps {
+    selectedItem: DropDownItem | undefined;
+    onClick?: React.EventHandler<any>;
+}
+
+export const DropDownLabel: React.SFC<DropDownLabelProps> = SBStateless(props => {
```
Comment (liorzisman) [inline src/components/drop-down/drop-down.tsx]: rename

File: src/components/drop-down/drop-down.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+export interface DropDownProps {
+    selectedItem: DropDownItem | undefined;
+    onLabelClick?: React.EventHandler<any>;
+}
+
+@SBComponent(style)
+export class DropDown extends React.Component<DropDownProps, {}> {
+    public render() {
+        return (
+            <div data-automation-id="DROP_DOWN" className="drop-down">
```
Comment (liorzisman) [inline src/components/drop-down/drop-down.tsx]: extends, root

File: src/components/drop-down/drop-down.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+          <CaretDown className="caret" />
+      </div>
+  );
+}, style);
+
+export interface DropDownItem {
+    label: string;
+}
+
+export interface DropDownProps {
```
Comment (liorzisman) [inline src/components/drop-down/drop-down.tsx]: extends, root

File: src/components/index.ts [hunk 7]
```diff
@@ -5,3 +5,4 @@ export * from './toggle';
 export * from './birthday-picker';
 export * from './checkbox';
 export * from './radio-group';
+export * from './drop-down';
```
Comment (liorzisman) [inline src/components/index.ts]: order

File: test/components/drop-down.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    const item = {
+        label: 'Test'
+    };
+
+    it('renders to the screen', () => {
+       const { select, waitForDom } = clientRenderer.render(<DropDown selectedItem={item} />);
+       return waitForDom(() => expect(select(dropDown)).to.be.present());
+    });
+
+    it('has default text initially', () => {
```
Comment (liorzisman) [inline test/components/drop-down.spec.tsx]: inaccurate

File: src/components/drop-down/drop-down.st.css (hunk 2 above)
Comment (liorzisman) [inline src/components/drop-down/drop-down.st.css]: renamed this confusing name :)

File: demo/components-demo.tsx (hunk 1 above)
Comment (liorzisman) [inline demo/components-demo.tsx]: added

**Comment by liorzisman** [comment]
@AviVahl ready for review

**Comment by liorzisman** [comment]
Approved by @AviVahl

**Pull request merged**